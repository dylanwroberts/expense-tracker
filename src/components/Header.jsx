import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
    <div>
        <h1>Expense Tracker</h1>
        <NavLink activeClassName="nav-link-active" to="/">Home</NavLink>
        <NavLink activeClassName="nav-link-active" to="/add">Add</NavLink>
    </div>
);

export default Header;