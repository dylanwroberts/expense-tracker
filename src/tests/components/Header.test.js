import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

test('should correctly render header', () => {
    const wrapper= shallow(<Header />);

    expect(wrapper).toMatchSnapshot();

    //expect the <Header/>'s h1's text to be 'Expense Tracker'
    expect(wrapper.find('h1').text()).toBe('Expense Tracker');

    // .find() acts similarly to how JQuery works - it allows you to
    // grab an item by its class, ID, tag type, etc...

    // .text() will return the string content from inside of an element
});