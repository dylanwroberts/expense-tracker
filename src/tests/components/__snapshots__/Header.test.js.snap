// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render header 1`] = `
<div>
  <h1>
    Expense Tracker
  </h1>
  <NavLink
    activeClassName="nav-link-active"
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="nav-link-active"
    to="/add"
  >
    Add
  </NavLink>
</div>
`;
